benchmark_runner/common/prometheus/prometeus_metircs.py

```diff
@@ -0,0 +1,34 @@
+
+from functools import wraps
+
+from benchmark_runner.common.logger.logger_time_stamp import logger_time_stamp, logger
+from benchmark_runner.common.prometheus.prometheus_metrics_operations import PrometheusMetricsOperation
+
+
+def prometheus_metrics(yaml_full_path: str = None):
```

Comment:
This looks to be a very complex way of doing it, although I do like that doing this as a decorator makes use very simple.

Reply:
Once it is in decorator, you just need to add the stamp "@prometheus_metrics" before a method.
There is option also to run it step by step:

```
    prometheus_metrics_operation = PrometheusMetricsOperation()
    prometheus_metrics_operation.init_prometheus()
    time.sleep(70)
    prometheus_metrics_operation.finalize_prometheus()
    prometheus_metrics_operation.run_prometheus_queries(query_yaml_file=yaml_full_path)
```